### PR TITLE
✨ Add support for passing GPP fields for amp-ad-network-doubleclick and adsense

### PR DESCRIPTION
**Pull request by zombifier**
WIP; this is dependent on #39789.

cc @powerivq

### CODE REVIEW

**Comment by powerivq** [comment]
@zombifier This PR should go out before https://github.com/ampproject/amphtml/pull/39789. 39789 would actually allow the GPP fields to be set by the CMPs and we want ads support of GPP to come out first.

**Comment by erwinmombay** [comment]
@powerivq i know you've explained this to me, but for posterity and documentation, why do we need ads support for GPP to come out first again?

**Comment by powerivq** [comment]
@erwinmombay I see some dependencies on https://github.com/ampproject/amphtml/pull/39789. W
e can merge it first if it is desirable.

**Comment by powerivq** [comment]
@zombifier That PR has been merged. You can rebase against the latest main to clean up the dependency.

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -642,17 +643,28 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       'gct': this.getLocationQueryParameterValue('google_preview') || null,
       'psts': tokens.length ? tokens : null,
       'gdpr': gdprApplies === true ? '1' : gdprApplies === false ? '0' : null,
-      'gdpr_consent':
-        consentStringType != CONSENT_STRING_TYPE.US_PRIVACY_STRING
-          ? consentString
-          : null,
+      'gdpr_consent': [
+        undefined,
```
Comment (powerivq) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js:647]: Is `undefined` needed here? Does it mean no consent strings at all?

Comment (zombifier) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js:647]: I was trying to preserve existing behavior of the `consentStringType != CONSENT_STRING_TYPE.US_PRIVACY_STRING` check, and there are existing unit tests of only TC String and no type that defaults to `gdpr_consent`. I think the intended behavior is defaulting to gdpr if a consent string with no type is found.

Comment (powerivq) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js:647]: SGTM

**Review by powerivq — APPROVED** [review]

**Comment by erwinmombay** [comment]
percy issue seems to be a flake? could you ammend and re-push to trigger the percy test again

**Pull request merged**